Makefile

```diff
@@ -1,11 +1,44 @@
+# The makefile for caffe. Pretty hacky.
+# $(info CXX_OBJS = $(CXX_OBJS))
 PROJECT := caffe
+COMMON_FLAGS += -std=c++11
 
-CONFIG_FILE := Makefile.config
-include $(CONFIG_FILE)
+# uncomment to build with debugging symbols
+DEBUG := 1
 
+# uncomment one
+CPU_ONLY := 1
```

Comment:
In the various Makefile.config.* files, CPU_ONLY := 1 is shown. This is confusing; what is the intended effect if CPU_ONLY is 1 in some and 0 in others? I think there should be a single authoritative source where the decision is made. I'm not sure how all this hangs together, but I'm thinking that since you introduced USE_OPENCL and USE_CUDA, CPU_ONLY should be removed. If neither USE_OPENCL nor USE_CUDA is specified, then what was called CPU_ONLY is what you are left with. I.e. one always gets the CPU implementation. USE_OPENCL and USE_CUDA will determine whether those implementations are available. But maybe you had something else in mind.